controllers/database.go

```diff
@@ -31,6 +36,52 @@ var dbTemplates = []string{
 	dbSecret,
 }
 
+// extract to var for mocking in testing
+var ConnectAndQueryDatabase = func(host, port, username, password, dbname string) bool {
+	connectionString := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, host, port, dbname)
+	db, err := sql.Open("mysql", connectionString)
+	if err != nil {
+		return false
+	}
+	defer db.Close()
```

Comment:
this block (with 'unhandled error/param' from calling `defer db.Close()`) is a common/ubiquitous db connection pattern, it would be more confusing to diverge from the norm here